src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -58,7 +91,7 @@ public async Task<IEnumerable<EmployerInfo>> GetLegalEntitiesForProviderAsync(lo
                     _logger.LogError(ex, $"Couldn't deserialise {nameof(ProviderPermissions)}.", null);
                 }
 
-                return await GetEmployerInfosAsync(providerPermissions);
+                return null;
```

Comment:
throw exception here perhaps as this is not a condition we expect and consuming code will have to handle this.

Reply:
I have refactored so all non-happy paths log an error and return an empty list of permissions.